### PR TITLE
Fix hashing and comparison for compression types

### PR DESCRIPTION
**Pull request by silviulica**
R: @robertwb Please take a look Robert. Thanks!

### CODE REVIEW

File: sdks/python/apache_beam/io/fileio.py [hunk 1]
```diff
@@ -283,7 +283,17 @@ def __init__(self, identifier):
...
+    return (isinstance(other, _CompressionType) and
+            self.identifier == other.identifier)
+
+  def __hash__(self):
+    return hash((self.identifier))
+
+  def __ne__(self, other):
+    return not self.__eq__(other)
+
+  def __str__(self):
```
Comment (robertwb) [inline sdks/python/apache_beam/io/fileio.py]: Make this __repr__ (and __str__ defaults to __repr__).

**Comment by robertwb** [comment]
I'm actually wondering if there's even any need for the CompressionType class. Maybe it should just be an enum? Either that or CompressionTypes should be unified with CompressionType with the static methods becoming instance methods.

**Comment by robertwb** [comment]
This PR has been subsumed by later changes, and can be closed.

**Comment by nssalian** [comment]
@silviulica could you please close this PR.

**Pull request closed without merging**